wireapp/wire-android
# fix(message): handle invalid embedded link in messages (WPB-3554)

----
#### PR Submission Checklist for internal contributors

- The **PR Title**

- The **PR Description**
----

# What's new in this PR?
Handle invalid schema links embedded in a message.
* Not crashing the app
* showing an error dialog
* 
### Issues

The app was crashing in case of invalid embedded links.

### Causes (Optional)

Link validity wasn't checked.

### Solutions

Check the link schema before opening it, and in case of being invalid we show a dialog.

### Testing
No UI tests.

#### Test Coverage (Optional)


#### How to Test

Send or receive a message with a wrong embedded link, when you click on the link, the app must not crash and an error dialog should appear.
![photo_2023-08-08 12 37 53](https://github.com/wireapp/wire-android-reloaded/assets/9512842/92caad6c-6415-4f89-b2d9-44e41209db71)


_Attachments like images, videos, etc. (drag and drop in the text box)_ 



----
#### PR Post Submission Checklist for internal contributors (Optional)

----
#### PR Post Merge Checklist for internal contributors

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.
